src/core/unit.h

```diff
@@ -957,6 +962,22 @@ int unit_thaw_vtable_common(Unit *u);
 #define log_unit_warning_errno(unit, error, ...) log_unit_full_errno(unit, LOG_WARNING, error, __VA_ARGS__)
 #define log_unit_error_errno(unit, error, ...)   log_unit_full_errno(unit, LOG_ERR, error, __VA_ARGS__)
 
+#define log_unit_struct_errno(unit, level, error, ...)                  \
+        ({                                                              \
+                const Unit *_u = (unit);                                \
```

Comment:
can you add local variables like this for level + error here too? You use both twice here, and we'd rather avoid double evaluation. adding local variables and only assigning them once means we avoid the double eval

Reply:
`error` appears in multiple places in these macros but is only eval'd once. You're right about `level`, though.